packages/model-viewer/src/test/features/animation-spec.ts

```diff
@@ -106,6 +106,39 @@ suite('ModelViewerElementBase with AnimationMixin', () => {
       });
     });
 
+    suite('when playOnce is invoked', () => {
+      setup(async () => {
+        const animationsPlay = waitForEvent(element, 'play');
+        element.playOnce();
+        await animationsPlay;
+      });
+
+      test('animations playOnce', async () => {
+        expect(animationIsPlaying(element)).to.be.true;
+      });
+
+      test('has a duration greater than 0', () => {
+        expect(element.duration).to.be.greaterThan(0);
+      });
+
+      suite('when pause is invoked', () => {
+        setup(async () => {
+          const animationsPause = waitForEvent(element, 'pause');
+          element.pause();
+          await animationsPause;
+        });
+
+        test('animations pause', () => {
+          expect(animationIsPlaying(element)).to.be.false;
+        });
+
+        test('changing currentTime triggers render', () => {
+          element.currentTime = 5;
+          expect(element[$scene].shouldRender()).to.be.true;
```

Comment:
Thanks for adding tests! One question: if `autoplay` is enabled and then `playOnce()` is  called (after any number of  loops), what is the result?  I think  I would expect the animation to continue to the end, then  pause. Perhaps we should add a test for that?

Reply:
I think so too about `autoplay` and `playOnce()`. I've added a test.
Please confirm that you have tested it correctly. (sorry for DeepL translation)